sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -386,15 +386,16 @@ def _get_small_trainset(self, X_binned_train, y_train, seed):
         with scorers.
         """
         subsample_size = 10000
-        rng = check_random_state(seed)
-        indices = np.arange(X_binned_train.shape[0])
         if X_binned_train.shape[0] > subsample_size:
-            # TODO: not critical but stratify using resample()
-            indices = rng.choice(indices, subsample_size, replace=False)
-        X_binned_small_train = X_binned_train[indices]
-        y_small_train = y_train[indices]
-        X_binned_small_train = np.ascontiguousarray(X_binned_small_train)
-        return X_binned_small_train, y_small_train
+            indices = np.arange(X_binned_train.shape[0])
+            indices = resample(indices, n_samples=subsample_size,
+                               replace=False, random_state=seed)
```

Comment:
If `self` is a classifier (`if is_classifier(self)`) and `y_train` is a set of integer class labels, `y_train` should be passed to the `stratify` kwarg of `resample`.

Reply:
Good catch!

I think that `y_train` is always encoded using `LabelEncoder()` when `self` is a classifier, so there's no need to check that check that `y_train` is a set of integer class labels:
https://github.com/scikit-learn/scikit-learn/blob/5310b87288fb5954814054530bccbd57692e1e08/sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py#L99

https://github.com/scikit-learn/scikit-learn/blob/5310b87288fb5954814054530bccbd57692e1e08/sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py#L949-L962